test/lib/reference_compare.py

```diff
@@ -0,0 +1,127 @@
+    def extract_passed_tests(self):
+        self.read()
+        
+        passed_tests_pattern = re.compile(
+            r"## Passed Tests\n\n\|Tag\|Test\|:clock1030: Duration\|Suite\|\n\|:---:\|:---:\|:---:\|:---:\|\n((\|.+\|\n)+)"
+        )
+
+        # Find the failed tests section
+        passed_tests_section = passed_tests_pattern.search(self.content)
+    
+        # Extract the passed tests table if it exists
+        if not passed_tests_section:
+            return []
+
+        passed_tests_table = passed_tests_section.group(1)
+    
+        # Regular expression to extract tag and test name from each row
+        # test_pattern = re.compile(r"\|(.+?)\|(.+?)\|.+?\|.+?\|.+?\|")
+        test_pattern = re.compile(r"(\|.+?\|.+?\|).+?\|.+?\|")
+
+        # Find all matches
+        passed_tests = test_pattern.findall(passed_tests_table)
+        return passed_tests
+
+def compare_summaries(markdown_file: str, reference_file: str):
+    md_file = MarkdownReader(markdown_file)
+    ref_file = MarkdownReader(reference_file)
+
+    md_summary = md_file.extract_summary()
+    ref_summary = ref_file.extract_summary()
+
+    if md_summary is None or ref_summary is None:
+        print("Comparison could not be performed due to missing summary data.", file=sys.stderr)
+        return False
+
+    error = 0
+    if md_summary != ref_summary:
+       print(f"error: Test results do not match the reference\n"
+          f"  Expected: Passed: {ref_summary.passed}, Failed: {ref_summary.failed}, Skipped: {ref_summary.skipped}, Total: {ref_summary.total}\n"
+          f"  Actual: Passed: {md_summary.passed}, Failed: {md_summary.failed}, Skipped: {md_summary.skipped}, Total: {md_summary.total}")
+       error = 1 # failure
+    
+    md_passed_tests = md_file.extract_passed_tests()
+    ref_passed_tests = ref_file.extract_passed_tests()
+
+    for testInfo in md_passed_tests:
+        if testInfo in ref_passed_tests:
+            ref_passed_tests.remove(testInfo)
+
+    if len(ref_passed_tests) > 0:
+        print(f"error: Regression detected. The following tests were expected to execute and pass:")
+        for reg_test in ref_passed_tests:
+            print(f"  {reg_test}")
+        error = 1 # failure
+
+    return error
+
+# def main():
+#     parser = argparse.ArgumentParser(description='Compare summaries in Markdown files.')
+#     parser.add_argument('-r', '--reference_file', type=str, required=True, help='Path to reference file')
+#     parser.add_argument('-m', '--markdown_file', type=str, default='summary_report.md', help='Path to consolidated summary markdown file')
+#     args = parser.parse_args()
+
+#     if compare_summaries(args.markdown_file, args.reference_file):
+#         print("Summaries are not equal.")
+#     else:
+#         print("Summaries are equal.")
+
+# if __name__ == '__main__':
+#     try:
+#         main()
+#     except Exception as e:
+#         print(f'An error occurred: {e}', file=sys.stderr)
+#         sys.exit(1)

```

Comment:
Is there a reason for this commented-out code or is it a left-over?

Reply:
This portion of the code was commented out to make the script debuggable. I have enabled it now, but it will not be used anywhere in the solution and is only for debugging purposes [Addressed review comments](https://github.com/Open-CMSIS-Pack/cmsis-toolbox/pull/155/commits/3457137cda2e15eaf7d669e1cbd37dcd96b63dbc)